### DISCUSSION TITLE
ここまでの振り返り

### DISCUSSION THREAD

**Discussion by wakamenori** [wakamenori/yuru-corpus-frontend#128] [General]
ここまでの開発を振り返り、ざっくりまとめる。

#  総括
確実に存在する問題に対して、ある程度適切なソリューションを提供できている。
一方で、その問題は想定より小さかったし、プロダクトのコンセプトも適切ではなかった。

# 課題設定
「あの話、どこでしてたっけ？」となった時、なかなか目的の回が見つからないというのを問題として捉えた。

## Good
- 確かに存在している課題である

例えばソシュールの話を探したいとき「ソシュール知ったかぶり講座」のシリーズは簡単に見つけられる。
だが、「赤ちゃんとオノマトペ」回でもソシュールの話がされていたことまでは分からない。さらに対象の動画を見つけても、30分以上の動画のどのタイミングで話していたかも分からない。
というわけで、既存の検索機能では解決できない課題であるのは確実。

## Bad

1. 規模が小さい
2. ペルソナの想定が甘い
3. サポコミュ外のリスナーの持つ感覚を掴めていない

### 1. 規模が小さい
検索ニーズは確実に存在するが規模が大きくない。サポコミュ内での「この話どの回かわかりますか？」系の質問も頻度は少ない。

### 2. ペルソナの想定が甘い
ここで想定しているユーザーは、一度動画を視聴済みであることを前提としている。ユーザーが欲している情報は「どの回で話していたか」であって、「どんな話をしていたか」ではない。
この点への認識が甘かった。別に文字起こし全文なんて誰も求めていない。

### 2. サポコミュ外のリスナーの持つ感覚を掴めていない
現時点で具体的に問題として現れているとは思っていないが、今後の設計が的外れになる可能性を孕んでいる。

# コンセプト
検索機能だけではプロダクトの規模が小さいと考え、少し大きくて包括的なコンセプトを設定した。
明文化するなら「既存メディアの代替」であり、Youtube/Spotifyの代わりにゆる言コーパスを利用するという想定をした。

## Good
特にいいことなし

## Bad
1. カスタマージャーニーを一切考えずに夢想した
2. 検索機能を最初に作らなかった

### 1. カスタマージャーニーを一切考えずに夢想した
ゆる言コーパスが文字起こしメディアとして機能するためには、いくつかの前提が必要となる。

一つ目の前提は「文字起こしを自主的かつ活発に修正する人が存在すること」である。
ゆる言コーパスの文字起こしデータはWhisperによって生成される。この状態からメディアとして成り立たせるには、人力での修正が絶対に必要になる。この作業を「放置しておいたらファンが勝手にやる」と想定してしまった。ゆる辞プロジェクトから何も学んでいない。
二つ目の前提は「ゆる言語学ラジオ関連コンテンツをテキストメディアで楽しみたい人がいること」である。
ゆる言語学ラジオのリスナー、特にサポーターコミュニティーに関しては、このような需要は非常に少ないといえる。なぜなら、ゆる言語学ラジオにおいて「パーソナリティー」に価値を見出している人が多いからである。
例えばニュース番組であれば記事で読んでもニュース番組で見ても殆ど価値は変わらないだろうが、「水野・堀元の知的な掛け合い」に価値を求めるリスナーにとって、単なるテキストデータは無機質になってしまうだろう。

### 2. 検索機能を最初に作らなかった
(これはコンセプトを間違えたことによる悪影響なので、少しレイヤーが違うが、ここにまとめておく。)

そもそも解決したい課題は既存の検索機能の弱さなのに、テキストメディアという誤ったコンセプトによって開発の優先順位が下がってしまった。
最初に文字起こしのCRUDを作ったのは完全にミス。

総じて、「既存メディアの代替」を目指したのはゆる言コーパスにおける最大の失敗

# UI/UX設計
## Good
1. 脱MUI
2. デザイン段階でDiscordに投げる

### 1. 脱MUI
一回MUIで作ったが、なんか気に入らないので辞めた。
結果として`styled-component`やcssを学ぶきっかけになったのでよかった。

### 2. デザイン段階でDiscordに投げる
サポーターコミュニティーに投げることで、自分の感覚だけでやらずに済んだ。

## Bad
1. デザインを作らないままでコードを描き始めることがあった
2. 配色を適当にやっている
3. 中途半端なMUI
4. 確定前に意見をもらうことが少ない
5. デザインの参考が少ない
6. UXが全体で繋がっていない

### 1. デザインを作らないままでコードを描き始めることがあった
ある程度イメージできたら「もうFigmaで作らなくてもいいや」となってしまう。
だが、その程度のものでも一回Figmaに書き出した方が絶対良い。

### 2. 配色を適当にやっている
いちいちハードコーディングしているのでテーマをContextで指定した方よい。

### 3. 中途半端なMUI
ヘッダーやボタンなど、各所でMUIの名残がある。きしょい。

### 4. 確定前に意見をもらうことが少ない
Goodに挙げたものの、もっと頻度を増やした方がよかった。また、プロトタイプでの共有も試してみるべきだった。

### 5. デザインの参考が少ない
[dribbble](https://dribbble.com/)を少し見た程度。特にUXはもっと色々見れたのでは？

### 6. UXが全体で繋がっていない
各機能のUXは明確だが、ユーザーがアクセスしてから離脱するまでが一切繋がっていない。

**Comment by dairappa** [comment]
なんかすっごい良いふりかえりだ。
